nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -1390,9 +1389,9 @@ impl DataStore {
                     &conn,
                     &opctx.log,
                     blueprint
-                        .all_omicron_zones_not_in(
-                            BlueprintZoneFilter::ShouldBeExternallyReachable,
-                        )
+                        .all_omicron_zones(|disposition| {
```

Comment:
I actually find this much clearer. And again, there is now a lower maintenance burden on both the code writing and reading side. You don't have to go look at the `BlueprintZoneFilter` to really understand what is happening under the hood.